maven-artifact/src/main/java/org/apache/maven/artifact/DefaultArtifact.java

```diff
@@ -314,7 +314,9 @@ public boolean equals(Object o) {
             return false;
         } else if (!a.getArtifactId().equals(artifactId)) {
             return false;
-        } else if (!a.getVersion().equals(version)) {
+        } else if (a.getVersion() != null && !a.getVersion().equals(version)) {
```

Comment:
How is that different from `!a.getGroupId().equals(groupId)` or `!a.getArtifactId().equals(artifactId)` ?

Reply:
So `validateIdentity` ensures some fields are non null or non empty.
This is not the case for `version` because the test is `(version == null) && (versionRange == null)`